src/ext/java/org/opentripplanner/ext/siri/SiriTimetableSnapshotSource.java

```diff
@@ -1027,7 +1026,7 @@ private Result<?, List<UpdateError>> handleModifiedTrip(
                 .ifFailure(errors::add);
             }
           } else {
-            buffer.update(pattern, tripTimes, serviceDate).ifFailure(errors::add);
+            cancelScheduledTrip(trip, serviceDate);
```

Comment:
Now the `cancelScheduledTrip(trip, serviceDate);` is also called in both of. the legs, should we move this outside the if and remove the else branch?